saleor/payment/gateways/braintree/plugin.py

```diff
@@ -0,0 +1,165 @@
+        },
+        "Use sandbox": {
+            "type": ConfigurationTypeField.BOOLEAN,
+            "help_text": pgettext_lazy(
+                "Plugin help text",
+                "Determines if Saleor should use Braintree sandbox API.",
+            ),
+            "label": pgettext_lazy("Plugin label", "Use sandbox"),
+        },
+        "Store customers card": {
+            "type": ConfigurationTypeField.BOOLEAN,
+            "help_text": pgettext_lazy(
+                "Plugin help text",
+                "Determines if Saleor should store cards on payments"
+                " in Braintree customer.",
+            ),
+            "label": pgettext_lazy("Plugin label", "Store customers card"),
+        },
+        "Automatic payment capture": {
+            "type": ConfigurationTypeField.BOOLEAN,
+            "help_text": pgettext_lazy(
+                "Plugin help text",
+                "Determines if Saleor should automaticaly capture payments.",
+            ),
+            "label": pgettext_lazy("Plugin label", "Automatic payment capture"),
+        },
+    }
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.config = None
+
+    @classmethod
+    def _get_default_configuration(cls):
+        defaults = None
+        return defaults
+
+    def _initialize_plugin_configuration(self):
+        super()._initialize_plugin_configuration()
+
+        if self._cached_config and self._cached_config.configuration:
+            configuration = self._cached_config.configuration
+
+            self.config = GatewayConfig(
+                gateway_name=GATEWAY_NAME,
+                auto_capture=configuration["Automatic payment capture"],
+                connection_params={
+                    "sandbox_mode": configuration["Use sandbox"],
+                    "merchant_id": configuration["Merchant ID"],
+                    "public_key": configuration["Public API key"],
+                    "private_key": configuration["Secret API key"],
+                },
+                template_path="",
+                store_customer=configuration["Store customers card"],
+            )
+        else:
+            # This should be removed after we drop payment configs in settings
+            gateway_config = settings.PAYMENT_GATEWAYS[GATEWAY_NAME]["config"]
+            self.config = GatewayConfig(
+                gateway_name=GATEWAY_NAME,
+                auto_capture=gateway_config["auto_capture"],
+                template_path=gateway_config["template_path"],
+                connection_params=gateway_config["connection_params"],
+                store_customer=gateway_config["store_card"],
+            )
+            self.active = GATEWAY_NAME in settings.CHECKOUT_PAYMENT_GATEWAYS
+
+    @classmethod
+    def _get_default_configuration(cls):
+        defaults = {
+            "name": cls.PLUGIN_NAME,
+            "description": "",
+            "active": False,
```

Comment:
Dynamic settings allow a user to enable or disable the plugin. Where do you handle it on the plugin side?

Reply:
Now all payment plugins have decorators to avoid running payment methods when disabled.